src/main/java/com/collicode/propertytracker/api/RoomApiController.java

```diff
@@ -24,13 +27,38 @@ public ResponseEntity<?> createRoom(@RequestHeader("X-RequestId") String request
       return ResponseEntity.status(500).body(ex.getMessage());
     }
   }
-  @GetMapping("/{apartmentId}")
+  @GetMapping("/{roomId}")
   public ResponseEntity<?> fetchRoomsByApartmentId(@RequestHeader("X-RequestId") String requestId,@PathVariable long apartmentId) {
     try {
       return ResponseEntity.ok().body(roomService.fetchRoomsByApartmentId(apartmentId));
     } catch (Exception ex) {
       return ResponseEntity.status(500).body(ex.getMessage());
     }
   }
+  @PutMapping("/{roomId}")
```

Comment:
while updating room add ApartmentId to be specific that you are updating a room in a particular appartment